### PR TITLE
Consistent catalog builders.

### PR DESCRIPTION
**Pull request by Cybermaxke**
[API ](https://github.com/SpongePowered/SpongeAPI/pull/1952) | **Common** | [Forge](https://github.com/SpongePowered/SpongeForge/pull/2565)

Updates PR https://github.com/SpongePowered/SpongeCommon/pull/2075 for bleeding.

### CODE REVIEW

File: testplugins/src/main/java/org/spongepowered/test/inventory/RecipeTest.java [hunk 1]
```diff
@@ -61,35 +65,63 @@
  *      adds a new recipe via a plugin (or any server where you simply add a new recipe). In the interest of not breaking testing altogether,
  *      disabling this plugin for now.
  */
-//@Plugin(id = "recipetest", name = "Recipe Test", description = "A plugin to test recipes")
+@Plugin(id = "recipe_test", name = "Recipe Test", description = "A plugin to test recipes")
```
Comment (ImMorpheus) [inline testplugins/src/main/java/org/spongepowered/test/inventory/RecipeTest.java]:
```suggestion
@Plugin(id = "recipe_test", name = "Recipe Test", description = "A plugin to test recipes", version = "0.0.0")
```

**Comment by ImMorpheus** [comment]
Search for `CatalogKey id`, may as well fix naming inconsistencies

See https://github.com/SpongePowered/SpongeCommon/pull/1792

**Comment by Cybermaxke** [comment]
@ImMorpheus I am not going to fix every name in this PR. Maybe for a separate one, or are you planning to update that PR?

**Comment by ImMorpheus** [comment]
Oh okay, I was assuming things since you did it for `SpongeDataRegistration`

I will update that PR.

**Pull request merged**